frontends/election-manager/src/hooks/use_election_manager_store.ts

```diff
@@ -0,0 +1,697 @@
+      );
+
+      await queryClient.invalidateQueries([electionDefinitionStorageKey]);
+      await queryClient.invalidateQueries([configuredAtStorageKey]);
+
+      await logger.log(
+        LogEventId.ElectionConfigured,
+        currentUserRoleRef.current,
+        {
+          disposition: LogDispositionStandardTypes.Success,
+          newElectionHash: newElectionDefinition.electionHash,
+        }
+      );
+    },
+    [logger, queryClient, reset, setStorageKeyAndLog]
+  );
+
+  const setCastVoteRecordFilesMutation = useMutation(
+    async (newCastVoteRecordFiles: CastVoteRecordFiles) => {
+      if (newCastVoteRecordFiles === CastVoteRecordFiles.empty) {
+        await fetch('/admin/write-ins/cvrs', { method: 'DELETE' });
+        await removeStorageKeyAndLog(cvrsStorageKey, 'Cast vote records');
+        await removeStorageKeyAndLog(
+          isOfficialResultsKey,
+          'isOfficialResults flag'
+        );
+      } else {
+        await setStorageKeyAndLog(
+          cvrsStorageKey,
+          newCastVoteRecordFiles.export(),
+          'Cast vote records'
+        );
+        await fetch('/admin/write-ins/cvrs', {
+          method: 'POST',
+          body: newCastVoteRecordFiles.export(),
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        });
+      }
+    },
+    {
+      onSuccess(data, newCastVoteRecordFiles) {
+        void queryClient.invalidateQueries([cvrsStorageKey]);
+        if (newCastVoteRecordFiles === CastVoteRecordFiles.empty) {
+          void queryClient.invalidateQueries([isOfficialResultsKey]);
+        }
+      },
+    }
+  );
+
+  const setCastVoteRecordFiles = useCallback(
+    async (newCastVoteRecordFiles: CastVoteRecordFiles) => {
+      await setCastVoteRecordFilesMutation.mutateAsync(newCastVoteRecordFiles);
+    },
+    [setCastVoteRecordFilesMutation]
+  );
+
+  const clearCastVoteRecordFiles = useCallback(async () => {
+    await setCastVoteRecordFiles(CastVoteRecordFiles.empty);
+  }, [setCastVoteRecordFiles]);
+
+  const markResultsOfficialMutation = useMutation(
+    async () => {
+      await logger.log(
+        LogEventId.MarkedTallyResultsOfficial,
+        currentUserRoleRef.current,
+        {
+          message:
+            'User has marked the tally results as official, no more Cvr files can be loaded.',
+          disposition: 'success',
+        }
+      );
+      await setStorageKeyAndLog(
+        isOfficialResultsKey,
+        true,
+        'isOfficialResults flag'
+      );
+    },
+    {
+      onSuccess() {
+        void queryClient.invalidateQueries([isOfficialResultsKey]);
+      },
+    }
+  );
+
+  const markResultsOfficial = useCallback(async () => {
+    await markResultsOfficialMutation.mutateAsync();
+  }, [markResultsOfficialMutation]);
+
+  const addPrintedBallotMutation = useMutation(
+    async (newPrintedBallot: PrintedBallot) => {
+      const oldPrintedBallots = await loadPrintedBallots();
```

Comment:
isn't this already cached? is there a reason to load it again?

Reply:
I can't just use the data from the `printedBallots` query since its data may not be available yet. Do you know of a better way to handle this?

Reply:
Hmm I see. I guess it's more straightforward when the backend API supports adding one new item.

I guess you could assert that the printed ballots data must be loaded before calling this mutation (which I would imagine is the case where it's used). Not sure it's a worthwhile optimization

Reply:
Looks like you did handle a similar case slightly differently below in `addFullElectionExternalTallyMutation`

Reply:
I'll unify them to both load when doing this. As you say, when we have an API that supports adding directly we can remove this.